src/steps/trackReplaceStep.ts

```diff
@@ -102,7 +104,9 @@ export function trackReplaceStep(
     in reference to !(fromA === fromB) - if changed ranges didnt change with that step, we need to insert at the start of the new range to match 
     where the user added inserted content
     */
-    const textWasDeleted = !!changeSteps.length && !(fromA === fromB)
```

Comment:
can you please check if  LEAN-4382 didnt regres as a result of this PR? Ideally, please add a test to make sure we dont regress.